sensorlib/src/main.py

```diff
@@ -0,0 +1,161 @@
+    :param noise_model_config: Noise model configuration as either a file name, JSON dictionary, or blank.
+                            - If the value is a valid .yaml file name, the file is loaded.
+                            - If the value is a JSON dictionary, the dictionary is used directly.
+                            - If passed as an empty string, default values are loaded from a configuration file.
+    :param detection_cycle_delay_seconds: Delay in sensor computation loop (seconds).
+    :param start_rpc_server: Starts an XML-RPC server if True.
+    :param xmlrpc_server_host: Host name for XML-RPC server.
+    :param xmlrpc_server_port: Port for XML-RPC server.
+
+    :return: No return if RPC server is running as this blocks execution; The SimulatedLidarSensor object if RPC server
+                is not running.
+    """
+    # Get inputs
+    sensor_transform = carla.Transform(carla.Location(x=0.0, y=0.0, z=0.0),
+                                       carla.Rotation(pitch=0.0, yaw=0.0, roll=0.0))
+
+    # Retrieve the sensor configuration
+    if os.path.exists(sensor_config):
+        config = SimulatedSensorUtils.load_config_from_file(sensor_config)
+    elif sensor_config is None or sensor_config == "":
+        config = SimulatedSensorUtils.load_config_from_file("../config/simulated_sensor_config.yaml")
+    else:
+        config = json.loads(sensor_config)
+    simulated_sensor_config = config["simulated_sensor"]
+    carla_sensor_config = config["lidar_sensor"]
+
+    # Retrieve noise model configuration
+    if os.path.exists(noise_model_config):
+        noise_model_config = SimulatedSensorUtils.load_config_from_file(noise_model_config)
+    if noise_model_config is None or noise_model_config == "":
+        noise_model_config = SimulatedSensorUtils.load_config_from_file("../config/noise_model_config.yaml")
+    else:
+        noise_model_config = json.loads(noise_model_config)
+
+    # Build sensor
+    simulated_lidar_sensor = SimulatedSensorConfigurator.register_simulated_semantic_lidar_sensor(
+        simulated_sensor_config,
+        carla_sensor_config,
+        noise_model_config,
+        infrastructure_id,
+        sensor_transform,
+        None)
+
+    # Compute detected objects continuously using a separate thread
+    scheduler = sched.scheduler(time.time, time.sleep)
+    scheduler.enter(detection_cycle_delay_seconds, 1, scheduled_compute, (scheduler, simulated_lidar_sensor,
+                                                                          detection_cycle_delay_seconds))
+    scheduler_thread = threading.Thread(target=scheduler.run)
+    print("Starting sensorlib compute.")
+    scheduler_thread.start()
+
+    # Create an XML-RPC server
+    if start_rpc_server:
+        print("Starting sensorlib XML-RPC server.")
+        server = SimpleXMLRPCServer((xmlrpc_server_host, xmlrpc_server_port))
```

Comment:
I think you might need to init a xmlrpc.server.SimpleXMLRPCRequestHandler, since this deals with the POST request

Reply:
Looks like this is automatic. Default parameter.